### PR TITLE
Adds Sentry test task to v3.0.x

### PR DESCRIPTION
**Pull request by halles**

### CODE REVIEW

**Comment by phptek** [comment]
Thanks for this. Please have a quick squiz at my comments for this same feature here: https://github.com/phptek/silverstripe-sentry/pull/29 - thanks!

File: src/Tasks/TestConnectionTask.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    /**
+     * Implement this method in the task subclass to
+     * execute via the TaskRunner
+     *
+     * @param \SilverStripe\Control\HTTPRequest|null $request
+     */
+    public function run($request = null)
+    {
+        foreach( Severity::ALLOWED_SEVERITIES as $severity) {
+            $response = captureMessage('Testing Severity Level: ' . $severity, new Severity($severity));
```
Comment (G-Rath) [inline src/Tasks/TestConnectionTask.php]: You can use [`NullTransport`](https://docs.sentry.io/platforms/php/#nulltransport) to have Sentry act like it sent events w/o actually sending them.

Comment (phptek) [reply src/Tasks/TestConnectionTask.php]: I've just realised that this is using Sentry's SDK directly. Shouldn't we be testing Sentry connections through the very same system that will report errors - the module logic itself?

I can make this change another time, it isn't a biggie. 

I've merged the 2.0 PR. Once the same changes are applied to this, I'll merge.

Comment (halles) [reply src/Tasks/TestConnectionTask.php]: @phptek cool. My aim was to test the connection, but I don't mind if it's done in a different way.

Comment (phptek) [reply src/Tasks/TestConnectionTask.php]: No worries, I don't want to burden you with anything further at this stage - I really appreciate the help :-) once the changes you guys have made to his task for the 2.0 release have bene made to this PR, I'm happy to merge.

**Comment by phptek** [comment]
@halles is this ready for re-review / merge yet?

**Comment by phptek** [comment]
This work has been incorporated separately into 4.0 (See: `5e0d18b8586`) and credited to @halles - thanks for your patience! ;-)

**Pull request closed without merging**